pkg/pool-utils/contracts/ProtocolFeeCache.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/BalancerErrors.sol";
+import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
+
+abstract contract ProtocolFeeCache {
+    uint256 public constant DELEGATE_PROTOCOL_FEES_SENTINEL = type(uint256).max;
```

Comment:
There is no need to make this public, as it is an implementation detail. We can keep it internal.

Reply:
I made it public so that factories (e.g., PhantomStablePool) could initialize it just using
ProtocolFeeCache.DELEGATE_PROTOCOL_FEES_SENTINEL()

Which is clearer than hard-coding a value.

Reply:
Ah, fair point!